### PR TITLE
fixed APIMANAGER-3402 and APIMANAGER-3513

### PR DESCRIPTION
**Pull request by kasun32**
When deleting an app earlier the confirmation message was same for all the apps with and without subscriptions. Edited the confirmation message to be more informative as shown below.

- When the app has subscriptions.
screenshot from 2015-04-27 14 52 59

- When the app doesn't have any subscriptions.
screenshot from 2015-04-27 14 53 20

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java [hunk 1]
```diff
@@ -3126,10 +3126,12 @@ public static NativeArray jsFunction_getApplications(Context cx,
             	 groupId = args[1].toString();
             }
              applications = apiConsumer.getApplications(new Subscriber(username), groupId);
+	        Subscriber subscriber = new Subscriber(username);
            
             if (applications != null) {
                 int i = 0;
                 for (Application application : applications) {
+	                int subscriptionCount = apiConsumer.getSubscriptionCount(subscriber,application.getName());
```
Comment (laki88) [inline components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java]: seems like wrong formatting

File: features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/fancy/templates/application/application-list/js/application.js [hunk 2]
```diff
@@ -126,9 +126,16 @@ function deleteApp(linkObj) {
     jagg.sessionAwareJS({redirect:'site/pages/applications.jag'});
     var theTr = $(linkObj).parent().parent();
     var appName = $(theTr).attr('data-value');
+    var apiCount = $(theTr).attr('api-count');
     $('#messageModal').html($('#confirmation-data').html());
-    $('#messageModal h3.modal-title').html(i18n.t('confirm.delete'));
-    $('#messageModal div.modal-body').text('\n\n'+i18n.t('confirm.deleteMsg')+'"' + appName + '"'+i18n.t('confirm.deleteMsgPostfix'));
+    if(apiCount > 0){
+    	$('#messageModal h3.modal-title').html(i18n.t('confirm.delete'));
+        $('#messageModal div.modal-body').text('\n\n' +i18n.t('confirm.deleteSubsMsg1')
```
Comment (laki88) [inline features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store/site/themes/fancy/templates/application/application-list/js/application.js]: seems like wrong formatting in here too

**Comment by kasun32** [comment]
@laki88 corrected the formatting errors. Thanks for pointing out.

**Comment by lakmali** [comment]
Did you test this with Application and Subscription grouping enabled? I think we have to modify the existing getSubscriptionCount method to get the subscription count when Apps and subscriptions are shared with a group.

**Comment by kasun32** [comment]
@lakmali no I didn't check for that scenario. Will check and do the necessary changes.

**Comment by kasun32** [comment]
@lakmali changed getSubscriptionCount to support Application and Subscription grouping. Please review and merge.

**Pull request merged**